src/main/java/sirius/kernel/commons/StringCleanup.java

```diff
@@ -316,6 +461,112 @@ public static String uppercase(@Nonnull String input) {
+        input = input.replace("&nbsp;", " ")
+                     .replace("&auml;", "ä")
+                     .replace("&ouml;", "ö")
+                     .replace("&uuml;", "ü")
+                     .replace("&Auml;", "Ä")
+                     .replace("&Ouml;", "Ö")
+                     .replace("&Uuml;", "Ü")
+                     .replace("&szlig;", "ß")
+                     .replace("&lt;", "<")
+                     .replace("&gt;", ">")
+                     .replace("&quot;", "\"")
+                     .replace("&apos;", "'")
+                     .replace("&amp;", "&")
+                     .replace("&#8226; ", "* ")
+                     .replace("&#8226;", "* ")
+                     .replace("&#8227; ", "* ")
+                     .replace("&#8227;", "* ")
+                     .replace("&#8259; ", "* ")
+                     .replace("&#8259;", "* ");
+        input = Strings.replaceAll(PATTERN_DECIMAL_ENTITY, input, s -> {
+            try {
+                return String.valueOf(Character.toChars(Integer.parseInt(s)));
+            } catch (NumberFormatException e) {
+                Exceptions.ignore(e);
+            } catch (Exception e) {
+                Exceptions.handle(e);
+            }
+            return "";
+        });
+        input = Strings.replaceAll(PATTERN_HEX_ENTITY, input, s -> {
+            try {
+                return String.valueOf(Character.toChars(Integer.parseInt(s, 16)));
+            } catch (NumberFormatException e) {
+                Exceptions.ignore(e);
+            } catch (Exception e) {
+                Exceptions.handle(e);
+            }
+            return "";
+        });
+
+        return input;
+    }
+
+    /**
+     * Normalizes a text by removing all HTML, entities and special characters.
+     *
+     * @param input the input to process
+     * @return the resulting string
+     */
+    @Nonnull
+    public static String htmlToPlainText(@Nonnull String input) {
+        String normalizedText = input;
+
+        if (STRIP_XML_REGEX.matcher(normalizedText).find()) {
+            // Reduce all contained whitespaces, tabs, and line breaks
+            normalizedText = Strings.cleanup(normalizedText, StringCleanup::reduceWhitespace);
+            // Replace br tags with line breaks
+            normalizedText = PATTERN_BR_TAG.matcher(normalizedText).replaceAll("\n");
+            // Replace li tags with line breaks
+            normalizedText = PATTERN_LILI_TAG.matcher(normalizedText).replaceAll("\n");
+            normalizedText = PATTERN_LI_TAG.matcher(normalizedText).replaceAll("\n");
+            // Replace p tags with line breaks
+            normalizedText = PATTERN_PP_TAG.matcher(normalizedText).replaceAll("\n");
+            normalizedText = PATTERN_P_TAG.matcher(normalizedText).replaceAll("\n");
+            // Remove any other tags
+            normalizedText = Strings.cleanup(normalizedText, StringCleanup::removeXml);
```

Comment:
I'd go for safeRemoveHtml here

Reply:
Why would I want to keep e.g. `<script>` tags in the plain representation of a text and have to worry to NEVER render it without escaping afterwards?

Reply:
As discussed, we stick to the more rigorous removal here